pkg/controller/spoketoken/spoke_token_controller.go

```diff
@@ -0,0 +1,250 @@
+// Copyright 2020 The Kubernetes Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package spoketoken
+
+import (
+	"context"
+	"encoding/json"
+	"time"
+
+	"github.com/pkg/errors"
+	corev1 "k8s.io/api/core/v1"
+	kerrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/rest"
+	"k8s.io/klog"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	"sigs.k8s.io/controller-runtime/pkg/source"
+
+	"github.com/open-cluster-management/multicloud-operators-subscription/pkg/utils"
+)
+
+const (
+	secretSuffix = "-cluster-secret"
+	requeuAfter  = 5
+)
+
+// Add creates a new agent token controller and adds it to the Manager if standalone is false.
+func Add(mgr manager.Manager, hubconfig *rest.Config, syncid *types.NamespacedName, standalone bool) error {
+	if !standalone {
+		hubclient, err := client.New(hubconfig, client.Options{})
+
+		if err != nil {
+			klog.Error("Failed to generate client to hub cluster with error:", err)
+			return err
+		}
+
+		return add(mgr, newReconciler(mgr, hubclient, syncid, hubconfig.Host))
```

Comment:
The managed cluster secret  `server` value should be managed cluster host endpoint. Here I see hub cluster host endpoint is assigned